test/integration/shoots/applications/shoot_app_test.go

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package applications
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"path/filepath"
+	"time"
+
+	. "github.com/gardener/gardener/test/integration/shoots"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/logger"
+
+	"github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	. "github.com/gardener/gardener/test/integration/framework"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	kubeconfig        = flag.String("kubeconfig", "", "the path to the kubeconfig  of the garden cluster that will be used for integration tests")
```

Comment:
Move your flag parsing and checking (checks for required / reasonable values) to an `init` function or do _all_ checks in `BeforeSuite` - when using your framework right now, it's actually possible to have a running test to a certain degree and then during cleanup stuff starts crashing because of a missing flag...

Reply:
I thought var blocks are executed before inits?

Reply:
Yes they are indeed, however your checks regarding those flags are running too late when the test is already ongoing. Your checks on your flags (e.g. whether a config value is set etc.) should be executed during an init-like method, just how we do it already for the gardener controller manager or how kubernetes does it with its Configuration.Complete pattern.

Reply:
there is a validation for the flags are now in their own function executed Before the test run.